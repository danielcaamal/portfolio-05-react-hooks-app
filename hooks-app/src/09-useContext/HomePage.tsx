import { useContext } from "react";
import { UserContext } from "./context/UserContext";


export const HomePage = () => {
    const { user, setUserContext } = useContext(UserContext);

    return (
        <>
            <h1
                id='home-page-title'
                >HomePage: {user?.name}</h1>
            <hr />
            <pre id='home-page-pre'>
                { JSON.stringify(user, null, 3) }
            </pre>
            <button
                id='home-page-button'
                type="button" 
                className="btn btn-primary"
                onClick={ () => setUserContext({ id: 123, name: 'John Doe', email: 'email@email.com' }) }
                >Set User
            </button>
        </>
    );
};