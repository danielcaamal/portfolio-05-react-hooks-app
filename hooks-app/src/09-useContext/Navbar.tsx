import { Link, NavLink } from "react-router-dom";

export const Navbar = () => {

    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark rounded-3">
            <div className="container-fluid">
                <Link className="navbar-brand" to="/">UseContext</Link>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavItem to="/">Home</NavItem>
                        <NavItem to="/login">Login</NavItem>
                        <NavItem to="/about">About</NavItem>
                    </div>
                </div>
            </div>
        </nav>
    );
}

const NavItem = ({ children, to } : {children: any, to:string}) => {
    return (
        <NavLink 
            className={ (args) => `nav-item nav-link ${args.isActive && 'active' }` }
            to={to}>
            {children}
        </NavLink>
    );
}