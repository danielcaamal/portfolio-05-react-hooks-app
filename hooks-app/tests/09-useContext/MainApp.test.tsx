import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { MainApp } from '../../src/09-useContext/MainApp';
import { BrowserRouter } from 'react-router-dom';


describe('MainApp', () => {
    it('should render', () => {
        const { container } = render(
            <MemoryRouter>
                <MainApp />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });

    it('should show HomePage', () => {
        const { container } = render(
            <MemoryRouter initialEntries={['/']}>
                <MainApp />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
        const title = container.querySelector('#home-page-title') as HTMLHeadingElement;
        expect(title.textContent).toContain('HomePage: ');
    });

    it('should show LoginPage', () => {
        const { container } = render(
            <MemoryRouter initialEntries={['/login']}>
                <MainApp />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
        const title = container.querySelector('#login-page-title') as HTMLHeadingElement;
        expect(title.textContent).toContain('LoginPage');
    });
});