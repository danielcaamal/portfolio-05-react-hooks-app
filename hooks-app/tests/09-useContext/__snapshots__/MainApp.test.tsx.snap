// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp should render 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark rounded-3"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        UseContext
      </a>
      <div
        class="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          class="navbar-nav"
        >
          <a
            aria-current="page"
            class="nav-item nav-link active"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-item nav-link false"
            href="/login"
          >
            Login
          </a>
          <a
            class="nav-item nav-link false"
            href="/about"
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>
  <hr />
  <h1
    id="home-page-title"
  >
    HomePage: 
  </h1>
  <hr />
  <pre
    id="home-page-pre"
  >
    {}
  </pre>
  <button
    class="btn btn-primary"
    id="home-page-button"
    type="button"
  >
    Set User
  </button>
</div>
`;

exports[`MainApp should show HomePage 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark rounded-3"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        UseContext
      </a>
      <div
        class="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          class="navbar-nav"
        >
          <a
            aria-current="page"
            class="nav-item nav-link active"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-item nav-link false"
            href="/login"
          >
            Login
          </a>
          <a
            class="nav-item nav-link false"
            href="/about"
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>
  <hr />
  <h1
    id="home-page-title"
  >
    HomePage: 
  </h1>
  <hr />
  <pre
    id="home-page-pre"
  >
    {}
  </pre>
  <button
    class="btn btn-primary"
    id="home-page-button"
    type="button"
  >
    Set User
  </button>
</div>
`;

exports[`MainApp should show LoginPage 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark rounded-3"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        UseContext
      </a>
      <div
        class="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          class="navbar-nav"
        >
          <a
            class="nav-item nav-link false"
            href="/"
          >
            Home
          </a>
          <a
            aria-current="page"
            class="nav-item nav-link active"
            href="/login"
          >
            Login
          </a>
          <a
            class="nav-item nav-link false"
            href="/about"
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>
  <hr />
  <h1
    id="login-page-title"
  >
    LoginPage
  </h1>
  <hr />
  <pre>
    {}
  </pre>
</div>
`;
